climakitae/explore/vulnerability.py

```diff
@@ -202,32 +252,70 @@ def cava_data(
 
-    print("--- Selecting Data Points --- \n")
+    time_new_line = "\n" if approach == "time" else ""
+    warm_new_line = "\n" if approach == "warming_level" else ""
+
+    print(f"--- Selecting Data Points --- {time_new_line}")
 
     data_pts = []
     for index, loc in input_locations.iterrows():
-
-        print(f"Selecting data for {loc['lat'], loc['lon']}")
         lat, lon = loc["lat"], loc["lon"]
 
-        selections = Select()
-        selections.data_type = "Gridded"
-        selections.downscaling_method = downscaling_method
-        selections.scenario_historical = [historical_data]
-        selections.scenario_ssp = clean_params["ssp_selected"]
-        selections.timescale = "hourly"
-        selections.variable_type = clean_params["variable_type"]
-        selections.variable = variable
-        selections.latitude = (
-            lat - 0.02,
-            lat + 0.02,
-        )  # Q: What padding to put for lat/lon values?
-        selections.longitude = (lon - 0.02, lon + 0.02)
-        selections.time_slice = (time_start_year, time_end_year)
-        selections.resolution = "3 km"
-        selections.units = units
-
-        data = selections.retrieve()
+        print(f"{warm_new_line}Selecting data for {lat, lon}")
+
+        if approach == "time":
+
+            selections = DataParameters()
+            selections.data_type = "Gridded"
+            selections.downscaling_method = downscaling_method
+            selections.scenario_historical = [historical_data]
+            selections.scenario_ssp = clean_params["ssp_selected"]
+            selections.timescale = "hourly"
+            selections.variable_type = clean_params["variable_type"]
+            selections.variable = variable
+            selections.latitude = (
+                lat - 0.02,
+                lat + 0.02,
+            )
+            selections.longitude = (lon - 0.02, lon + 0.02)
+            selections.time_slice = (time_start_year, time_end_year)
+            selections.resolution = "3 km"
+            selections.units = units
+
+            data = selections.retrieve()
+
+        elif approach == "warming_level":
+
+            wl = warming_levels()
+            wl.wl_params.timescale = "hourly"
+            wl.wl_params.downscaling_method = downscaling_method
+            wl.wl_params.variable_type = "Derived Index"
+            wl.wl_params.variable = "NOAA Heat Index"
+            wl.wl_params.latitude = (lat - 0.02, lat + 0.02)
+            wl.wl_params.longitude = (lon - 0.02, lon + 0.02)
+            wl.wl_params.warming_levels = [
+                warming_level
+            ]  # Calvin- default, only allow for 1 warming level to be passed in.
+            wl.wl_params.units = units
+            wl.wl_params.resolution = "3 km"
+            wl.wl_params.anom = "No"  # Q: When do we want this anomaly to be 'Yes'?
+
+            wl.calculate()
```

Comment:
Note to self: This should incorporate some sort of `get_closest_gridcell` functionality here, since the above params are designed to grab an individual gridcell.